xbmc/pvr/PVRComponentRegistration.h

```diff
@@ -0,0 +1,22 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "pvr/IPVRComponent.h"
+#include "utils/ComponentContainer.h"
+
+namespace PVR
+{
+class CPVRComponentRegistration : public CComponentContainer<IPVRComponent>
+{
+public:
+  CPVRComponentRegistration();
+  virtual ~CPVRComponentRegistration();
```

Comment:
```c++
~CPVRComponentRegistration() override;
```

Reply:
@garbear base class has no virtual dtor.

Reply:
Shouldn't it have one?

Reply:
it's not necessary with shared_ptr's as they carry deleters which don't rely on virtual dtors, assuming things are created using make_shared and not as
```c++
 std::shared_ptr<base> foo; 
foo.reset(new Derived);
```